src/printer_tree.c

```diff
@@ -4337,3 +4418,131 @@ tree_print_parsed_submodule(struct ly_out *out, const struct lysp_submodule *sub
 
     return erc;
 }
+
+/* @param ext[in] context allocated for extension */
+/* @param refs[out] set of lysc nodes matching parent-refernce xpaths? */
+static LY_ERR
+trb_parent_references(const struct lysc_ext_instance *ext, struct ly_set **refs)
```

Comment:
Looking at this again, please put this function directly into `schema_mount.c` and make it publicly available similarly to `lyplg_ext_schema_mount_create_context()`, you can call it `lyplg_ext_schema_mount_get_parent_ref()`. Also, it is already implemented there, in function `schema_mount_dup_parent_ref()` so it would be best if you could refactor it to not duplicate the code.

Reply:
OK, I will move that to schema_mount.c and refactor so that the xpath evaluation is shared, but schema_mount_dup_parent_ref() duplicates data nodes and there is no data context when printing the tree.  There may not be a huge savings resulting from the refactoring.

Reply:
Hm, you are right. But please move the function there anyway, I would say that file is more suitable because it is schema-mount functionality that is not specific to the tree printer and could theoretically be used for something else in the future.